### PR TITLE
Value adjuster component

### PR DESCRIPTION
**Pull request by rdjpalmer**
## What

Value adjuster component to allow development of the [gift funnel](https://trello.com/c/VyL1oTHx)

## Todo

- [x] Icon vs value sizing
- [x] Style-able buttons (any thoughts on this? Pass in `btnClassName`?)

## Design

screen shot 2016-02-01 at 14 09 51

### Function

value-adjuster

### Default style

screen shot 2016-02-01 at 14 47 15

### CODE REVIEW

File: components/ValueAdjuster/ValueAdjuster.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ValueAdjuster.propTypes = {
+  value: PropTypes.number.isRequired,
+  onChange: PropTypes.func.isRequired,
+
+  min: PropTypes.number,
+  max: PropTypes.number,
+  prefix: PropTypes.string,
+  suffix: PropTypes.string,
+  changeAmount: PropTypes.number,
+  className: PropTypes.string,
```
Comment (iest) [inline components/ValueAdjuster/ValueAdjuster.js]: Perhaps a `btnClassName` would be good too?

File: components/ValueAdjuster/ValueAdjuster.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import Icon from 'components/Icon/Icon';
+import ScreenReadable from 'components/ScreenReadable/ScreenReadable';
+
+import css from './ValueAdjuster.css';
+
+export default class ValueAdjuster extends Component {
+  constructor(props) {
+    super(props);
+    this.incrementValue = this.incrementValue.bind(this);
+    this.decrementValue = this.decrementValue.bind(this);
```
Comment (iest) [inline components/ValueAdjuster/ValueAdjuster.js]: Why not arrow functions?

**Comment by iest** [comment]
Looks good so far. Why have it in loggins?

File: components/ValueAdjuster/ValueAdjuster.js (hunk 2 above)
Comment (rdjpalmer) [inline components/ValueAdjuster/ValueAdjuster.js]: Loggins was shouting at me for using arrow functions ¯\_(ツ)_/¯

**Comment by rdjpalmer** [comment]
Loggins, 'cause its on par with something like the slider im

**Comment by iest** [comment]
Fair enough.

Why doesn't loggins allow new ES stuff? We should fix that. Need to give it a lot of love...

**Comment by rdjpalmer** [comment]
@iest Updated the PR with the remaining tasks.

RE new ES stuff, probably something to do with when we upgraded to react 0.14 and downgraded again?

**Comment by iest** [comment]
Great overall, just one thing.

Instead of having `prefix` and `suffix`, why not have a `transformValue` callback passed in? Means we'd have a load more flexibility:

```jsx
const prefix = value => `£${value}`;
<ValueAdjuster transform={prefix} ... />

const transform = value => value * 5;
<ValueAdjuster transform={transform} ... />

const GBPToUSD = amount => amount * 0.7
const transform = value => `£${value} ($${GBPToUSD(value)})`;
<ValueAdjuster transform={transform} ... />
```

**Comment by iest** [comment]
Way better :clap: 
:ship:

**Pull request merged**